src/App.tsx

```diff
@@ -1,14 +1,48 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>();
+  const [status, setStatus] = useState('all');
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const handleCloseButton = () => {
+    setSelectedTodo(undefined);
+  };
+
+  useMemo(() => {
+    setVisibleTodos(
+      todos.filter(todo => {
+        return (
+          (status === 'completed'
+            ? todo.completed
+            : status === 'active'
+              ? !todo.completed
+              : todo) && todo.title.toLowerCase().includes(query.toLowerCase())
+        );
+      }),
+    );
+  }, [status, todos, query]);
```

Comment:
wrong useMemo usage
```suggestion
  const [query, setQuery] = useState('');

  useEffect(() => {
    setLoading(true);
    getTodos()
      .then(setTodos)
      .finally(() => setLoading(false));
  }, []);

  const handleCloseButton = () => {
    setSelectedTodo(undefined);
  };

  const visibleTodos = useMemo(() => {
    // TODO add filtarion logic. Note: no need to filter in case status is all and query is empty.
    // Also do not use nesting ternary
  }, [status, todos, query]);
```